### PR TITLE
S9 loading state

### PR DESCRIPTION
**Pull request by NunoDasNeves**
Loading state pretty much everywhere its needed now.
Some other optimizations to reduce the number of times courses get loaded from api.
Might have gone overboard on that buuut I think it's ok. Thoughts?

### CODE REVIEW

File: frontend/src/store/root.js [hunk 1]
```diff
@@ -1,37 +1,54 @@
...
 }
 
 const mutations = {
-  SET_LOADING(state, bool) {
+  TOGGLE_LOADING(state, bool) {
     state.loading = bool
   },
-  POPULATE_SEARCH(state, data) {
-    state.courses = data
+  REFRESH_COURSES(state, courses) {
```
Comment (DarkPurple141) [inline frontend/src/store/root.js]: This is for what, a review update?

Comment (NunoDasNeves) [reply frontend/src/store/root.js]: Yeah, atm the idea is to load courses once only then update whenever an individual course is loaded...

**Comment by NunoDasNeves** [comment]
@darkpurple141 added transitions. I think it's marginally better ... Not sure what else to do

**Comment by NunoDasNeves** [comment]
Oh there's conflicts...

File: frontend/src/App.vue [hunk 2]
```diff
@@ -149,4 +150,12 @@ textarea:focus, textarea:active {
   border: 1px solid #acc;
 }
 
+.fade-slide-enter-active {
+  transition: all 0.5s;
```
Comment (DarkPurple141) [inline frontend/src/App.vue]: transition: prop time ease-in-out, prop time ease-in-out;

**Pull request merged**